dialer_test.go

```diff
@@ -343,18 +344,33 @@ func TestDialerResolver(t *testing.T) {
 				"example.com": {"127.0.0.1"},
 			},
 		},
+		{
+			scenario: "resolve domain with port to ip without port",
+			address:  "example.com:9092",
+			resolver: map[string][]string{
+				"example.com": {"127.0.0.1:"},
+			},
+		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.scenario, func(t *testing.T) {
-			topic := makeTopic()
-			createTopic(t, topic, 1)
-			defer deleteTopic(t, topic)
-
 			d := Dialer{
 				Resolver: &mockResolver{addrs: test.resolver},
 			}
 
+			if resolverResult, _ := lookupHost(context.TODO(), test.address, d.Resolver); resolverResult != "127.0.0.1:9092" {
+				t.Errorf("bad resolver result, got %s expected %s", resolverResult, "127.0.0.1:9092")
+			}
+
+			if !runCompleteTest {
+				return
+			}
```

Comment:
In my opinion the test should be split into a "real" unit test, which only checks the resolver behaviour and and a integration test which connects to kafka.
I don't know if this makes sense because allof the tests are "integration tests" and another behaviour would break the coding style.

Your decision:
- shall i remove my "uint-test-code"
- shall i completley split up the test in integration/unit-test

Reply:
Adding a separate test to validate the resolver behavior sounds fine to me, are there any downsides or blockers?

Reply:
Hello @themulle, did you get a chance to look at addressing this last suggestion?

Reply:
@themulle up? Let me know if you want us to take this over as well, no rush :)